File: 1806-count-of-matches-in-tournament/count-of-matches-in-tournament.py
class Solution:
    def numberOfMatches(self, n: int) -> int:
        # code here

        ans = 0
        while n > 1:
            if n%2 != 0: # if no of teams is odd
                ans += int(n//2) 
                n = int(n//2) +1
            else:
                ans += int(n/2)
                n = int(n/2)
        return ans

# time complexity: o(log N)
# space complexity: o(log N)

File: 1806-count-of-matches-in-tournament/count-of-matches-in-tournament.cpp
class Solution {
public:
    int numberOfMatches(int n) {
        int ans = 0;
        while (n > 1) {
        if (n % 2 != 0) {
            ans += n / 2;
            n = n / 2 + 1;
        } else {
            ans += n / 2;
            n /= 2;
        }
        }
        return ans;
    }
};

// time complexity: O(log N)
// space complexity: O(1)